Compile-time error on MacOS

The library can't be used in macOS due to a missing `fuse.ENODATA` symbol.

Is it possible to replace it with something go-generic, like `os.ErrNotExists`?

```
# github.com/CalebQ42/squashfs
../../../go/pkg/mod/github.com/!caleb!q42/squashfs@v0.7.8/fuse2.go:103:19: undefined: fuse.ENODATA
../../../go/pkg/mod/github.com/!caleb!q42/squashfs@v0.7.8/fuse2.go:115:14: undefined: fuse.ENODATA
../../../go/pkg/mod/github.com/!caleb!q42/squashfs@v0.7.8/fuse3.go:101:19: undefined: fuse.ENODATA
../../../go/pkg/mod/github.com/!caleb!q42/squashfs@v0.7.8/fuse3.go:113:14: undefined: fuse.ENODAT
```

--- a/fuse2.go
+++ b/fuse2.go
@@ -100,7 +100,7 @@ func (f fileNode2) ReadAll(ctx context.Context) ([]byte, error) {
 		_, err := f.WriteTo(&buf)
 		return buf.Bytes(), err
 	}
-	return nil, fuse.ENODATA
+	return nil, ENODATA
 }
 
 func (f fileNode2) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.ReadResponse) error {
@@ -112,7 +112,7 @@ func (f fileNode2) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.R
 		}
 		return nil
 	}
-	return fuse.ENODATA
+	return ENODATA
 }
 
 func (f fileNode2) ReadDirAll(ctx context.Context) (out []fuse.Dirent, err error) {

--- a/fuse3.go
+++ b/fuse3.go
@@ -98,7 +98,7 @@ func (f fileNode) ReadAll(ctx context.Context) ([]byte, error) {
 		_, err := f.WriteTo(&buf)
 		return buf.Bytes(), err
 	}
-	return nil, fuse.ENODATA
+	return nil, ENODATA
 }
 
 func (f fileNode) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.ReadResponse) error {
@@ -110,7 +110,7 @@ func (f fileNode) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.Re
 		}
 		return nil
 	}
-	return fuse.ENODATA
+	return ENODATA
 }
 
 func (f fileNode) ReadDirAll(ctx context.Context) (out []fuse.Dirent, err error) {

--- a/fuse_darwin.go
+++ b/fuse_darwin.go
@@ -1,5 +1,3 @@
-//go:build !linux
-
 package squashfs
 
 import (

--- a/fuse_darwin.go
+++ b/fuse_darwin.go
@@ -1,0 +1,9 @@
+//go:build !linux
+
+package squashfs
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+var ENODATA = unix.Errno(unix.ENODATA)

--- a/fuse_linux.go
+++ b/fuse_linux.go
@@ -1,0 +1,3 @@
+package squashfs
+
+var ENODATA = fuse.ENODATA

--- a/fuse_windows.go
+++ b/fuse_windows.go
@@ -1,0 +1,3 @@
+package squashfs
+
+var ENODATA = windows.Errno(windows.ENODATA)

--- a/go.mod
+++ b/go.mod
@@ -13,5 +13,5 @@ require (
 
 require (
 	github.com/seaweedfs/fuse v1.2.2
-	golang.org/x/sys v0.2.0 // indirect
+	golang.org/x/sys v0.6.0 // indirect
 )

